### PR TITLE
Add G-UNI

### PR DESCRIPTION
**Pull request by kassandraoftroy**

### CODE REVIEW

File: .guni.env [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# DEFAULT_GAS_PRICE=  50 GWEI by default
+
+TOKEN=0x50379f632ca68D36E50cfBC8F78fe16bd1499d1e
+ATOKEN=0x517AD97cD3543eE616cDb3D7765b201D6c9dFFdd
+STABLE_DEBT_TOKEN=0x135bb9dfd7880a53ef86b6f281aa4c3a9addb85c
+VARIABLE_DEBT_TOKEN=0x104e375e7a62ac88317b93a2288865513c1dc511
+INTEREST_STRATEGY=0xfffE32106A68aA3eD39CcCE673B646423EEaB62a
```
Comment (miguelmtzinf) [inline .guni.env]: incorrect?  the one placed at the proposal is: 0x52E39422cd86a12a13773D86af5FdBF5665989aD

**Comment by miguelmtzinf** [comment]
@kassandraoftroy Would be nice if you remove/un-commit the artifacts changes

File: contracts/assetListing/AssetListingGUni.sol [hunk 2]
```diff
@@ -0,0 +1,155 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity 0.7.5;
+pragma abicoder v2;
+
+import {IAaveOracle} from "./interfaces/IAaveOracle.sol";
+
+interface IPoolConfigurator {
```
Comment (miguelmtzinf) [inline contracts/assetListing/AssetListingGUni.sol]: Could you rename it so its clear is the one of the AMM Market?

File: contracts/assetListing/AssetListingGUni.sol [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+interface IProposalIncentivesExecutor {
+  function execute() external;
+}
+
+contract AssetListingGUni is IProposalIncentivesExecutor {
+  address constant POOL_CONFIGURATOR = 0x23A875eDe3F1030138701683e42E9b16A7F87768;
+  address constant AAVE_ORACLE = 0xA50ba011c48153De246E5192C8f9258A2ba79Ca9;
+
+  // BPT V2 to list
```
Comment (miguelmtzinf) [inline contracts/assetListing/AssetListingGUni.sol]: Could you rewrite the comments and replace `BPT` with `GUNI` please? Same with `Balancer` and so

File: test/test-listing-guni.spec.ts [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+
+const DAI_TOKEN = '0x6b175474e89094c44da98b954eedeac495271d0f';
+const DAI_HOLDER = '0x72aabd13090af25dbb804f84de6280c697ed1150';
+
+const ERRORS = {
+  NO_BORROW: '7',
+  NO_COLLATERAL_BALANCE: '9',
+  NO_STABLE_BORROW: '12',
+};
+
```
Comment (miguelmtzinf) [inline test/test-listing-guni.spec.ts:56]: need to load the `.guni.env` to make it work

File: test/test-listing-guni.spec.ts [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+    await (await gov.execute(proposal)).wait();
+    
+    const proposalState = await gov.getProposalState(proposal);
+    expect(proposalState).to.be.equal(7);
+    const {
+      configuration: { data },
+      aTokenAddress,
+      stableDebtTokenAddress,
+      variableDebtTokenAddress,
+    } = await pool.getReserveData(GUNI1);
```
Comment (miguelmtzinf) [inline test/test-listing-guni.spec.ts]: could you add tests for all the assets the proposal is listing?

File: test/test-listing-guni.spec.ts [hunk 6]
```diff
@@ -0,0 +1,233 @@
...
+    console.log("guni holder borrows dai")
+    await (
+      await pool.connect(guniHolder).borrow(dai.address, parseEther('1'), 2, 0, GUNI_HOLDER)
+    ).wait();
+
+    // proposer NOT able to borrow G-UNI
+    const borrowAmount = parseEther('10').div(decimalMultiplier);
+    await expect(
+      pool.connect(proposer).borrow(guni.address, borrowAmount, 2, 0, proposer.address)
+    ).to.be.reverted;
```
Comment (miguelmtzinf) [inline test/test-listing-guni.spec.ts]: Would be nice to uset `revertedWith` and check the revert message

File: test/test-listing-guni.spec.ts [hunk 7]
```diff
@@ -0,0 +1,233 @@
...
+
+    await expect(
+      pool.borrow(guni.address, borrowAmount, 1, 0, proposer.address)
+    ).to.be.reverted;
+    increaseTime(40000);
+  });
+
+  it("Oracle should return a non zero G-UNI price", async () => {
+    expect(await oracle.getAssetPrice(GUNI1)).to.be.gt('0')
```
Comment (miguelmtzinf) [inline test/test-listing-guni.spec.ts]: check oracle addition for all the assets getting listed please

**Review by miguelmtzinf — CHANGES_REQUESTED** [review]

File: .guni.env (hunk 1 above)
Comment (kassandraoftroy) [reply .guni.env]: fixed

File: contracts/assetListing/AssetListingGUni.sol (hunk 2 above)
Comment (kassandraoftroy) [reply contracts/assetListing/AssetListingGUni.sol]: 👍

File: contracts/assetListing/AssetListingGUni.sol (hunk 3 above)
Comment (kassandraoftroy) [reply contracts/assetListing/AssetListingGUni.sol]: 👍

File: test/test-listing-guni.spec.ts (hunk 4 above)
Comment (kassandraoftroy) [reply test/test-listing-guni.spec.ts:56]: added above (L27-L36)

File: test/test-listing-guni.spec.ts (hunk 5 above)
Comment (kassandraoftroy) [reply test/test-listing-guni.spec.ts]: yes, thats true - added

File: test/test-listing-guni.spec.ts (hunk 7 above)
Comment (kassandraoftroy) [reply test/test-listing-guni.spec.ts]: done

File: test/test-listing-guni.spec.ts (hunk 6 above)
Comment (kassandraoftroy) [reply test/test-listing-guni.spec.ts]: cool - added.

File: .guni.env (hunk 1 above)
Comment (kassandraoftroy) [reply .guni.env]: also note that most of these env variables are not currently being used (they are hardcoded constants in the contract) so they can also be removed if you like.

File: test/test-listing-guni.spec.ts [hunk 8]
```diff
@@ -0,0 +1,277 @@
...
+  TOKEN0,
+  TOKEN1
+} = process.env;
+
+if (!TOKEN0 || !TOKEN1) {
+  throw new Error('You have not set correctly the .env file');
+}
+
+const AAVE_GOVERNANCE_V2 = "0xEC568fffba86c094cf06b22134B23074DFE2252c"
+const AAVE_LENDING_POOL = "0x7937D4799803FbBe595ed57278Bc4cA21f3bFfCB" //'0x7d2768dE32b0b80b7a3454c06BdAc94A69DDc7A9';
```
Comment (miguelmtzinf) [inline test/test-listing-guni.spec.ts]: could remove the comment please?

File: test/test-listing-guni.spec.ts [hunk 9]
```diff
@@ -0,0 +1,277 @@
...
+      proposer
+    )) as IERC20;
+    await (await guni0.connect(guniHolder).approve(pool.address, parseEther('200000'))).wait();
+
+    // GUNI deposit by guni holder
+    const depositedAmount0 = parseEther('3');
+    await (
+      await pool.connect(guniHolder).deposit(guni0.address, depositedAmount0, GUNI_HOLDER, 0)
+    ).wait();
+    expect(await aGuni.balanceOf(GUNI_HOLDER)).to.gte(depositedAmount0.sub(1));
```
Comment (miguelmtzinf) [inline test/test-listing-guni.spec.ts]: why not use `.to.eq`?

File: test/test-listing-guni.spec.ts [hunk 10]
```diff
@@ -0,0 +1,277 @@
...
+      await pool.connect(guniHolder).borrow(dai.address, parseEther('1'), 2, 0, GUNI_HOLDER)
+    ).wait();
+
+    reserveData = await pool.getReserveData(TOKEN1);
+    poolData = parsePoolData(reserveData.configuration.data);
+    expect(poolData).to.be.eql(expectedConfig);
+
+    // preparing for tests on G-UNI DAI/USDC
+    aGuni = (await ethers.getContractAt('IERC20', reserveData.aTokenAddress, proposer)) as IERC20;
+    stableDebt = (await ethers.getContractAt('IERC20', reserveData.stableDebtTokenAddress, proposer)) as IERC20;
```
Comment (miguelmtzinf) [inline test/test-listing-guni.spec.ts]: debtTokens are not needed I guess

**Review by miguelmtzinf — CHANGES_REQUESTED** [review]

File: test/test-listing-guni.spec.ts (hunk 10 above)
Comment (kassandraoftroy) [reply test/test-listing-guni.spec.ts:232]: I use them later in the test to make sure that borrowing is disabled

File: test/test-listing-guni.spec.ts (hunk 8 above)
Comment (kassandraoftroy) [reply test/test-listing-guni.spec.ts]: removed

File: test/test-listing-guni.spec.ts (hunk 9 above)
Comment (kassandraoftroy) [reply test/test-listing-guni.spec.ts]: haha agree! (was just copying pattern from FRAX pr)

File: test/test-listing-guni.spec.ts (hunk 10 above)
Comment (kassandraoftroy) [reply test/test-listing-guni.spec.ts]: removed

**Review by miguelmtzinf — APPROVED** [review]
looks good to me now, thanks for the contribution!

**Pull request closed without merging**